### ISSUE TITLE
Augmented Steam Widget Error

### ISSUE THREAD

**Issue by Revadike** [tfedor/AugmentedSteam#463]
Implement the features of https://greasyfork.org/en/scripts/39492-steam-error-widget-helper by @Ryzhehvost

What the author said:
![image](https://user-images.githubusercontent.com/4411977/63272717-da05a300-c29c-11e9-99ab-67cf5881f624.png)

**Comment by Revadike** [comment]
For this to work properly, we can't rely on just `/api/appdetails?appids=` API as this often still doesn't return anything. Instead, AS should implement a product info API, that requests the product info of an app (like `print_app_info APPID` in steam console). I know how to do this in node.js. It is also possible in C#.

**Comment by Ryzhehvost** [comment]
Full-fledged steam client on server side just to get app name seems overkill. But I found good api that may help:
```
https://api.steampowered.com/IStoreService/GetAppList/v1/?key=${steam_api_key}&last_appid=${appid-1}&max_results=1
```
Problem is - it requires steam api key, and putting one to extension is unacceptable. We may make a "proxy" on a server side, so that AS server will get needed info with api key, and return to extension without one. But, on the other hand, that would mean anyone can use this api without api key, and this may be not wanted. Then again, if we will cache this data on AS server it won't do much harm even if someone will use it.
And one more thing - before using this API - we need to test it, maybe it has some limitations too (than we can combine this api and store api I used in my userscript). So we should test it for some time before actually doing anything.

**Comment by Revadike** [comment]
Of course not a real steam client. A virtualized one: check SteamKit.
Do you think applist API will return better results than appdetails API? I doubt it.

**Comment by Ryzhehvost** [comment]
SteamKit is exactly what I meant by "Full-fledged steam client". As per this api - as I said above, I don't know if it's better or not; It worked with appid where appdetails failed, that's all I know now. I will test it. Even if it's not perfect - we may combine those two api to get acceptable results.

**Comment by tfedor** [comment]
is this meant for widgets that you can add on other pages?

**Comment by Revadike** [comment]
> is this meant for widgets that you can add on other pages?

Other pages? It's meant for these kind of widgets:
![image](https://user-images.githubusercontent.com/4411977/64057453-3ae68280-cb9d-11e9-8f12-ba9cbb8dab10.png)

**Comment by tfedor** [comment]
but thats used outside of Stean, right?

**Comment by Revadike** [comment]
> but thats used outside of Steam, right?

No, I don't think so, and if it was, I think it would be via an iframe.

**Comment by tfedor** [comment]
Im not sure where I can find thia widget then, on community pages? Can you link me please?

**Comment by Revadike** [comment]
> Im not sure where I can find thia widget then, on community pages? Can you link me please?

Steam will automatically convert steam app store links posted in steam community to this widget

![image](https://user-images.githubusercontent.com/4411977/64057513-2f478b80-cb9e-11e9-8b33-20955acde7d5.png)

**Comment by Ryzhehvost** [comment]
It's used both in steam community and "outside" of steam, but in both cases that's an iframe.

**Comment by Ryzhehvost** [comment]
Update: I tested this API, and it won't solve the problem, even together with appdetails there are still cases that are not covered.
So, it seems there is only two options left:
Either install steam client aka SteamKit on server and make own database of appid -> name pairs with open API for AS (as @Revadike suggested above), or fetch and parse html from community page. And I actually don't like both ways 😞

**Comment by Revadike** [comment]
I think having a steam database would be great for AS's backend. Applications could be many. Even just for app names, it's fine. There isn't really an alternative anyway. Steam community doesn't always work, either. Valve can terminate the steam community page too. Not to mention that scaping info from HTML is a bad practice for this kind of things.

**Comment by Ryzhehvost** [comment]
Well, if it depends on backend - there is nothing to do here for me. In case someone will be doing this - here is some links for testing:
https://store.steampowered.com/widget/836620/?dynamiclink=1
https://store.steampowered.com/widget/1109770/?dynamiclink=1
https://store.steampowered.com/widget/513780/?dynamiclink=1
https://store.steampowered.com/widget/757950/?dynamiclink=1
https://store.steampowered.com/widget/603780/?dynamiclink=1
https://store.steampowered.com/widget/648590/?dynamiclink=1

**Comment by Revadike** [comment]
> there is nothing to do here for me

You could already write code without retrieving the name part.

**Comment by Revadike** [comment]
In case anyone was wondering: The new library API doesn't work either: https://store.steampowered.com/api/libraryappdetails/?appid=648590&l=english

Actually, it works for some, like 007 Legends:
https://store.steampowered.com/widget/211670/?dynamiclink=1
https://store.steampowered.com/api/libraryappdetails/?appid=211670&l=english
